### ISSUE TITLE
[Bug] User is presented with the same error regardless of the API error

### ISSUE THREAD

**Issue by olivierobert** [pakaponk/gsr-scraping#17]
## Issue

When signing up for an account, the error "This email has already been used" regardless of the API error:

![image](https://user-images.githubusercontent.com/696529/197107422-7e23fa57-cd9e-4712-84b3-6f8f648ffab0.png)

This is very confusing when setting up the application and for any new user 😅 

Upon review of the code, it makes sense since the error has been hard-coded:

https://github.com/pakaponk/gsr-scraping/blob/940933277b8173378865961d48de952baf8f99e4/apps/web/src/Components/SignupForm.tsx#L59-L67

STR:
1. In a terminal, `cd app/web/` and run `nom run dev` (so running on the web application, not the API 😉 This allows to simulate an API issue easily)
2. In a web browser, go to `http://localhost:3000/` and click on `create a new account`
3. Fill in the form and click on "Sign up".

## Expected

The user should be presented with a valid error message depending on the issue: already used email address, password not following security requirements, API error, etc.

**Comment by pakaponk** [comment]
Yep. Totally agree.

There is a couple of things to address this issue.

1. we need to differentiate the error from `fetch` and the Error response from our API to allow us to display an appropriate message for each error.
2. at Nest.js, we need to refactor `/auth/local/register` to throw different kinds of errors, "already used email address" or "password not following security requirement" as you suggested, now all of them are thrown as Internal Server Error.

The first point can be addressed alongside Issue #18. Would you mind if I addressed that issue first, then resolved this one later? 
Note: I don't mind resolving this one first before addressing issue #18, we simply have to add a temporary hotfix to `fetchSignup` to throw different kinds of errors for `fetch` and Error Response.

**Comment by olivierobert** [comment]
Indeed, working on the API responses is a pre-requisite for this bug 👍
